// admin-bro/lib/frontend/components/property-type/richtext/edit.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactDom = require("react-dom");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _designSystem = require("../../design-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const toolbarOptions = [[{
  header: [1, 2, 3, 4, 5, 6, false]
}], ['bold', 'italic', 'underline', 'strike'], // toggled buttons
['blockquote', 'code-block'], [{
  list: 'ordered'
}, {
  list: 'bullet'
}], [{
  script: 'sub'
}, {
  script: 'super'
}], // superscript/subscript
[{
  indent: '-1'
}, {
  indent: '+1'
}], // indent
[{
  direction: 'rtl'
}], // text direction
[{
  size: ['small', false, 'large', 'huge']
}], // custom dropdown
[{
  color: []
}, {
  background: []
}], // dropdown with defaults from theme
[{
  font: []
}], [{
  align: []
}], ['clean'] // remove formatting button
];

const Wrapper = _styledComponents.default.div.attrs({
  className: 'control has-icons-right'
}).withConfig({
  displayName: "edit__Wrapper",
  componentId: "sc-1ilg3d7-0"
})([".ql-toolbar{border-color:", ";.ql-picker{color:", ";}}.ql-container{border-color:", ";background:", ";}"], ({
  theme
}) => theme.colors.grey40, ({
  theme
}) => theme.colors.grey60, ({
  theme
}) => theme.colors.grey40, ({
  theme
}) => theme.colors.white);

const loadQuill = () => new Promise(resolve => {
  const id = 'quill-script-tag';

  if (window.document.getElementById(id)) {
    // it could be a situation where id exists but quill hasn't been loaded. In this case
    // we check if Quill global variable exists
    const checkIfLoaded = () => {
      if (typeof Quill === 'function') {
        resolve();
      }
    };

    checkIfLoaded();
    setInterval(checkIfLoaded, 500);
    return;
  }

  const script = window.document.createElement('script');
  script.src = 'https://cdn.quilljs.com/1.3.6/quill.js';
  script.async = true;
  script.defer = true;
  script.id = id;
  script.addEventListener('load', () => {
    resolve();
  });
  const style = window.document.createElement('link');
  style.rel = 'stylesheet';
  style.type = 'text/css';
  style.href = 'https://cdn.quilljs.com/1.3.6/quill.snow.css';
  window.document.body.appendChild(script);
  window.document.body.appendChild(style);
});

class Edit extends _react.default.Component {
  constructor(props) {
    super(props);
    this.wysiwigRef = /*#__PURE__*/_react.default.createRef();
  }

  componentDidMount() {
    loadQuill().then(() => {
      this.setupWysiwig();
    });
  }

  shouldComponentUpdate(nextProps) {
    const {
      record,
      property
    } = this.props;

    if (!nextProps) {
      return false;
    }

    const oldError = record.errors && record.errors[property.name] && record.errors[property.name].message;
    const newError = nextProps.record.errors && nextProps.record.errors[property.name] && nextProps.record.errors[property.name].message;
    return oldError !== newError;
  }

  componentDidUpdate() {
    this.setupWysiwig();
  }

  setupWysiwig() {
    const {
      property,
      record
    } = this.props;
    const value = record.params && record.params[property.name] || '';
    this.wysiwigRef.current.innerHTML = value;

    if (this.quill) {
      delete this.quill; // eslint-disable-next-line react/no-find-dom-node

      const thisNode = (0, _reactDom.findDOMNode)(this);
      const toolbars = thisNode.getElementsByClassName('ql-toolbar');

      for (let index = 0; index < toolbars.length; index += 1) {
        toolbars[index].remove();
      }
    }

    this.quill = new Quill(this.wysiwigRef.current, _objectSpread({
      modules: {
        toolbar: toolbarOptions
      },
      theme: 'snow'
    }, property.custom));
    this.quill.on('text-change', () => {
      this.handleChange(this.wysiwigRef.current.children[0].innerHTML);
    });
  }

  handleChange(value) {
    const {
      onChange,
      property
    } = this.props;
    onChange(property.name, value);
  }

  render() {
    const {
      property,
      record
    } = this.props;
    const error = record.errors && record.errors[property.name];
    return /*#__PURE__*/_react.default.createElement(_designSystem.FormGroup, {
      error: !!error
    }, /*#__PURE__*/_react.default.createElement(_designSystem.Label, {
      htmlFor: property.name,
      required: property.isRequired
    }, property.label), /*#__PURE__*/_react.default.createElement(Wrapper, null, /*#__PURE__*/_react.default.createElement("div", {
      className: "quill-editor",
      ref: this.wysiwigRef,
      style: {
        height: '400px'
      }
    })), /*#__PURE__*/_react.default.createElement(_designSystem.FormMessage, null, error && error.message));
  }

}

exports.default = Edit;
